src/Util/FileUploader.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Util;
+
+use Gedmo\Sluggable\Util\Urlizer;
+use phpDocumentor\Reflection\DocBlock\Tags\Var_;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Request;
+
+class FileUploader
+{
+    /**
+     * The directory where uploaded files will be placed.
+     *
+     * @var string
+     */
+    private $uploadDirectory;
+
+    /**
+     * The directory where file chunks will be stored.
+     *
+     * @var string
+     */
+    private $chunksDirectory;
+
+    public function __construct(string $homedirPrefix)
+    {
+        $this->uploadDirectory = $homedirPrefix . DIRECTORY_SEPARATOR . 'upload' . DIRECTORY_SEPARATOR . 'files';
+        $this->chunksDirectory = $homedirPrefix . DIRECTORY_SEPARATOR . 'upload' . DIRECTORY_SEPARATOR . 'chunks';
+    }
+
+    /**
+     * File uploader service.
+     *
+     * @param Request $request Symfony request instance.
+     *
+     * @return array
+     */
+    public function upload(Request $request): array
+    {
+        dump($request);
+        $totalChunks = $request->get('dztotalchunkcount');
+        $chunkIndex = $request->get('dzchunkindex');
+        /** @var UploadedFile $uploadedFile */
+        $uploadedFile = $request->files->get('file');
+        $originalFilename = pathinfo($uploadedFile->getClientOriginalName(), PATHINFO_FILENAME);
+        $fileExtension = $uploadedFile->guessExtension();
+        $newFilename = Urlizer::urlize($originalFilename).'-'.uniqid().'.'.$fileExtension;
+        $uuid = $request->get('dzuuid');
+        $targetDirectory = '';
+        $fileSize = $uploadedFile->getSize();
+
+        if ($totalChunks > 1) {
+            $chunksFolder = $this->chunksDirectory . DIRECTORY_SEPARATOR . $uuid;
+            $this->isFolder($chunksFolder);
+            $isChunk = true;
+            $uploadedFile->move(
+                $chunksFolder,
+                $chunkIndex
+            );
+
+            if ((int)$totalChunks === ($chunkIndex + 1)) {
```

Comment:
You can't assume that the chunkIndex will be in order. It is possible that the last chunk is very small and will finish first.

Reply:
So, we should check the size of the file uploaded with the total size of the actual file? or something like that along with this?

Reply:
We should probably check the number of chunks in the UUID folder, and if the last part completes, then assemble the chunks.